provider/internal/java/provider.go

```diff
@@ -173,6 +173,13 @@ func (p *javaProvider) ProviderInit(ctx context.Context) error {
 }
 
 func (p *javaProvider) Init(ctx context.Context, log logr.Logger, config provider.InitConfig) (provider.ServiceClient, error) {
+	//By default if nothing is set for analysis mode, in the config, we should default to full for external providers
+	var a provider.AnalysisMode = provider.AnalysisMode(config.AnalysisMode)
+	if a == provider.AnalysisMode("") {
+		a = provider.FullAnalysisMode
+	} else if !(a == provider.FullAnalysisMode || a == provider.ParitalAnalysisMode || a == provider.SourceOnlyAnalysisMode) {
```

Comment:
Probably move this check to a common helper function?

Reply:
I don't think it is necessary, soon the only place this code will live is in the provider-server as java will move to an external provider as well.